xarray/core/variable.py

```diff
@@ -406,6 +407,96 @@ def __getitem__(self, key):
         return type(self)(dims, values, self._attrs, self._encoding,
                           fastpath=True)
 
+    def _broadcast_indexes(self, key):
+        """
+        Parameters
+        -----------
+        key: One of
+            array
+            a mapping of dimension names to index.
+
+        Returns
+        -------
+        dims: Tuple of strings.
+            Dimension of the resultant variable.
+        indexers: list of integer, array-like, or slice. This is aligned
+            along self.dims.
+        """
+        if not utils.is_dict_like(key):
+            key = {self.dims[0]: key}
+        example_v = None
+        indexes = OrderedDict()
+        for k, v in key.items():
+            if not isinstance(v, (integer_types, slice, Variable)):
+                if not hasattr(key, 'ndim'):  # convert list or tuple
+                    v = np.array(v)
+            if example_v is None and isinstance(v, Variable):
+                example_v = v
+            indexes[k] = v
+
+        # When all the keys are array or integer, slice
```

Comment:
Yes, `y=[0, 1]` -> `Variable(['y'], [1, 0])`.

I think the result here would look like `Variable(['a', 'y'], [[3, 4], [0, 1]])`:
- After making them broadcast compatible, the indexer variables have dimensions `['a', 'y']` and shapes `(2, 1)`/`(1, 2)`.
- For the `x` coordinate, each coordinate value is taken in order.
- For the `y` coordinate, the first two coordinates are taken in reverse order.
- The result is labeled by the dimensions of the indexers (i.e., `('a', 'y')`).